benchmark/Speed.hs

```diff
@@ -136,8 +138,14 @@ samplesBenchmarks lrData hmmData ldaData = benchmarks
           m <- models
           return (s, m, a)
 
-speedLengthCSV :: FilePath
-speedLengthCSV = "speed-length.csv"
+normalBenchmarks :: [Benchmark]
+normalBenchmarks = [ bench "Normal single sample monad bayes" $ nfIO $ do
+                       sampleIOfixed (do xs <- replicateM 1000 $ normal 0.0 1.0
+                                         return $ sum xs)
+                   , bench "Normal single sample monad bayes fu" $ nfIO $ do
+                       FU.sampleIOfixed (do xs <- replicateM 1000 $ normal 0.0 1.0
+                                            return $ sum xs)
```

Comment:
You are calling polymorphic code (the inner `do` blocks) on particular types. This means that additionally to the actual computation, type class dictionary lookup is also performed. This can have a performance impact. Maybe this can be improved by adding a `SPECIALISE` pragma to the `MonadDistribution` instance definitions of both sampler modules. See https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/pragmas.html#specialize-pragma.

I imagine a change like this:

```haskell
  uniform a b = SamplerT (ReaderT $ uniformRM (a, b))
  {-# SPECIALISE uniform :: Double -> Double -> SamplerIO Double #-}
```
This is for the MWC sampler, and a similar pragma can be added to the `random-fu` sampler. Hopefully, both will be faster and more comparable then.